Src/main.c

```diff
@@ -99,15 +100,27 @@ int main(void)
   /* USER CODE BEGIN 2 */
   debug("\n\nBootcamp starting up...");
   debug("Compiled on %s at %s", __DATE__, __TIME__);
-
+  HAL_ADC_Start(&hadc);
+  HAL_TIM_PWM_Start(&htim16,TIM_CHANNEL_1);
 
   HAL_GPIO_WritePin(LED1_GPIO_Port, LED1_Pin, GPIO_PIN_SET);
   /* USER CODE END 2 */
 
   /* Infinite loop */
   /* USER CODE BEGIN WHILE */
+  int minDutyCycleMs = 1;
+  int maxDutyCycleMs = 2;
+  int differenceDutyCycleMs = maxDutyCycleMs - minDutyCycleMs;
+  int arrPeriod = 60000; //arrPeriod in this context is equal to the maxDutyCycleMs
```

Comment:
I missed this in my last review. You'll want to double check this value; you sure it corresponds to maxDutyCycleMs ?

ALso a great example of the sort of thing unit testing can catch :p. Part of the reason we've starting doing that on the ZeroPilot.

Reply:
I think so, since 60000/(3x10^6) = 0.02, which is what we loaded up into the Arr period register? I may be wrong tho lol

Reply:
0.02s is 20 ms. your'e right that that's the period, but the duty cycle needs to always be between 1 and 2 ms.